### PR TITLE
[SPARK-21287][SQL] Ability to use Integer.MIN_VALUE as a fetchSize

### PR DESCRIPTION
**Pull request by maver1ck**
## What changes were proposed in this pull request?

FIX for https://issues.apache.org/jira/browse/SPARK-21287

## How was this patch tested?

Existing automated tests + manual tests.

### CODE REVIEW

**Comment by srowen** [comment]
See JIRA; I don't think we should allow this.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala [hunk 1]
```diff
@@ -103,10 +103,11 @@ class JDBCOptions(
       s" and '$JDBC_NUM_PARTITIONS' are required.")
   val fetchSize = {
     val size = parameters.getOrElse(JDBC_BATCH_FETCH_SIZE, "0").toInt
-    require(size >= 0,
+    require(size >= 0 || size == Integer.MIN_VALUE,
```
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala]: It does not make sense to the other data sources except MySQL. If needed, we can introduce a dialect specific checking APIs in [JdbcDialect.scala](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala)

Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala]: +1 for @gatorsmile 's suggestion.

**Comment by gatorsmile** [comment]
@maver1ck  Any update?

**Pull request closed without merging**